src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,291 @@
+            }
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var operation = semanticModel.GetOperation(declarationSyntax, cancellationToken) as IVariableDeclarationGroupOperation;
+            if (operation?.Declarations.Length != 1)
+            {
+                return default;
+            }
+
+            var localDeclaration = operation.Declarations[0];
+            if (localDeclaration.Declarators.Length != 1)
+            {
+                return default;
+            }
+
+            var declarator = localDeclaration.Declarators[0];
+            if (declarator.Initializer is null)
+            {
+                return default;
+            }
+
+            var localType = declarator.Symbol?.Type;
+            if (localType is null)
+            {
+                return default;
+            }
+
+            if (declarator.Initializer.Value is null ||
+                declarator.Initializer.Value.Kind == OperationKind.Invalid)
+            {
+                return default;
+            }
+
+            var isValidSelection = await CodeRefactoringHelpers.RefactoringSelectionIsValidAsync(
+                document,
+                selection,
+                node: declarationSyntax,
+                holes: ImmutableArray.Create(declarator.Initializer.Value.Syntax),
+                cancellationToken).ConfigureAwait(false);
+
+            if (!isValidSelection)
+            {
+                return default;
+            }
+
+            if (!IsLegalUsingStatementType(semanticModel.Compilation, localType))
+            {
+                return default;
+            }
+
+            return (declarationSyntax, declarator.Symbol);
+        }
+
+        private static bool IsLegalUsingStatementType(Compilation compilation, ITypeSymbol type)
+        {
+            var disposableType = compilation.GetSpecialType(SpecialType.System_IDisposable);
+            if (disposableType is null)
+            {
+                return default;
+            }
+
+            // CS1674: type used in a using statement must be implicitly convertible to 'System.IDisposable'
+            return compilation.ClassifyCommonConversion(type, disposableType).IsImplicit;
+        }
+
+        private async Task<Document> IntroduceUsingStatementAsync(
+            Document document,
+            TextSpan span,
+            CancellationToken cancellationToken)
+        {
+            var (declarationStatement, localVariable) = await FindDisposableLocalDeclaration(document, span, cancellationToken).ConfigureAwait(false);
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            if (!(declarationStatement.Parent is TBlockSyntax parent))
+            {
+                throw new NotImplementedException();
```

Comment:
this needs to be handled.  either by checking up front that we're in a block and not offering the fix if we're not.  or by having us handle all the things that can parent a locl-decl statement.

Reply:
Yes. This is my way of blocking out WIPs (see the todos list in the comment I linked).